openmetadata-ui/src/main/resources/ui/src/utils/EntityLineageUtils.tsx

```diff
@@ -62,18 +76,46 @@ import {
 } from '../enums/entity.enum';
 import { SearchIndex } from '../enums/search.enum';
 import { AddLineage, EntitiesEdge } from '../generated/api/lineage/addLineage';
+import { Container } from '../generated/entity/data/container';
+import { Dashboard } from '../generated/entity/data/dashboard';
+import { Mlmodel } from '../generated/entity/data/mlmodel';
+import { SearchIndex as SearchIndexEntity } from '../generated/entity/data/searchIndex';
+import { Column, Table } from '../generated/entity/data/table';
+import { Topic } from '../generated/entity/data/topic';
 import { ColumnLineage, LineageDetails } from '../generated/type/entityLineage';
 import { EntityReference } from '../generated/type/entityReference';
 import { addLineage, deleteLineageEdge } from '../rest/miscAPI';
-import {
-  getPartialNameFromFQN,
-  getPartialNameFromTableFQN,
-} from './CommonUtils';
+import { getPartialNameFromTableFQN } from './CommonUtils';
 import { getEntityName } from './EntityUtils';
+import Fqn from './Fqn';
 import { showErrorToast } from './ToastUtils';
 
 export const MAX_LINEAGE_LENGTH = 20;
 
+export const encodeLineageHandles = (handle: string) => {
+  return btoa(encodeURIComponent(handle));
+};
+
+export const decodeLineageHandles = (handle: string) => {
+  return decodeURIComponent(atob(handle));
+};
+
+export const getColumnSourceTargetHandles = (obj: {
+  sourceHandle?: string | null;
+  targetHandle?: string | null;
+}) => {
+  const { sourceHandle, targetHandle } = obj;
+
+  return {
+    sourceHandle: sourceHandle
+      ? decodeLineageHandles(sourceHandle)
+      : sourceHandle,
```

Comment:
Can we move this condition to util fun itself?
```suggestion
    sourceHandle: decodeLineageHandles(sourceHandle)
```